### PR TITLE
Add a helper to handle seasonality via regression

### PR DESCRIPTION
**Pull request by fritzo**
Addresses #2311 

This adds a `periodic_features()` helper that can be used to generate covariates for forecasting models.

## Tested
- [x] added shape and bound tests

### CODE REVIEW

File: pyro/ops/tensor_utils.py [hunk 1]
```diff
@@ -109,6 +109,39 @@ def periodic_cumsum(tensor, period, dim):
...
+    treated via regression, and short seasonality can be treated via the
+    :func:`periodic_repeat` and :func:`periodic_cumsum` helpers. For example,
+    to combine yearly with weekly seasonality one could set
+    ``max_period=365.25`` and ``min_period=7``.
+
+    :param int duration: Number of discrete time steps.
+    :param float max_period: Optional max period, defaults to ``duration``.
+    :param float min_period: Optional min period (exclusive), defaults to 2 = Nyquist cutoff.
+    :param \*\*options: Tensor construction options, e.g. ``dtype`` and ``device``.
+    :returns: A ``(duration, 2 * ceil(max_period - min_period) - 2)``-shaped tensor of features
```
Comment (fehiepsi) [inline pyro/ops/tensor_utils.py]: nit: `2 * ceil(max_period - min_period) - 2` -> `2 * floor(max_period / min_period)`

Comment (fritzo) [reply pyro/ops/tensor_utils.py]: oops, will fix docs. but I'm pretty sure `ceil(x) - 1 != floor(x)` and that we need `ceil` here. Am I mistaken?

Comment (fehiepsi) [reply pyro/ops/tensor_utils.py]: nope, you are right. `arange(1, 3.)` returns `1, 2`

Comment (fritzo) [reply pyro/ops/tensor_utils.py]: fixed.

**Comment by fritzo** [comment]
Thanks for your careful review, @fehiepsi !

**Pull request merged**